lib/scope.js

```diff
@@ -235,6 +235,7 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
                     ref.reference(options);
                 });
                 node.thedef = def;
+                node.reference(options);
```

Comment:
... you are confusing me.

It's not the first time `AST_Symbol.reference()` is used - it's the only way to register any `AST_SymbolRef` to its `SymbolDef`, and to mark its presence along `AST_Scope` leaf nodes for `mangle` to function properly.

https://github.com/mishoo/UglifyJS2/blob/a5ffe2c23fdfaf13f3466a01d9dd9d590c5e8672/lib/scope.js#L209